client/components/edit/create/CreateObject.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import { useFetchContext } from "../../../context/FetchContext";
+import { apiUrl, newEditObjectUrl } from "../../../util/routes";
+import Link from "next/link";
+import ParentControl from "../../shared/ParentControl";
+import NoParentControl from "../../shared/NoParentControl";
+import StateControls from "../../shared/StateControls";
+import CategoryTreeView from "./CategoryTreeView";
+
+interface CreateObjectProps {
+    parentPid: string;
+    allowChangeParentPid: boolean;
+    allowNoParentPid: boolean;
+}
+
+const CreateObject = ({
+    parentPid = "",
+    allowNoParentPid = false,
+    allowChangeParentPid = true,
+}: CreateObjectProps): React.ReactElement => {
+    const {
+        action: { fetchText, fetchJSON },
+    } = useFetchContext();
+    const [models, setModels] = useState({});
+    const [selectedModel, setSelectedModel] = useState("");
+    const [results, setResults] = useState({
+        state: "initial",
+        content: "",
+    });
+    const [title, setTitle] = useState("");
+    const [currentState, setCurrentState] = useState("Inactive");
+    const [noParent, setNoParent] = useState(allowNoParentPid && parentPid === "");
+    const [parent, setParent] = useState(parentPid);
+    const handleParentChange = (event) => {
+        setParent(event.target.value);
+        if (allowNoParentPid) {
+            setNoParent(event.target.value.length === 0);
+        }
+    };
+
+    const handleNoParentChange = (event) => {
+        setNoParent(event.target.checked);
+        if (allowChangeParentPid && event.target.checked) {
+            setParent("");
+        }
+    };
+
+    useEffect(() => {
+        async function loadModels() {
+            let models = {};
+            try {
+                models = await fetchJSON(apiUrl + "/edit/models");
+            } catch (e) {
+                const error = e as Error;
+                console.error("Problem fetching models: ", error.message);
+            }
+            setModels(models);
+        }
+        loadModels();
+    }, []);
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        setResults({
+            state: "loading",
+            content: "Working...",
+        });
+        try {
+            const params = {
+                method: "POST",
+                body: JSON.stringify({
+                    title,
+                    parent,
+                    model: selectedModel,
+                    state: currentState,
+                }),
+            };
+            const headers = { "Content-Type": "application/json" };
+            const pid = await fetchText(newEditObjectUrl, params, headers);
+            setResults({
+                state: "loading",
+                content: pid,
```

Comment:
The original CreateObject component displayed an "object created" message followed by a link to the newly-created PID upon success. The refactored component just outputs the PID by itself in an unlinked state.

I would definitely like to see the UI improved so that CreateObject opens in a lightbox and reports success through the SnackBar. Is this an incremental step in that direction, or an oversight? I'm nearly ready to approve this PR, but this feels like a loss of functionality. If there's not a simple way to fix it here, I'm happy to merge as-is if we immediately follow up with another PR to restore the functionality in an improved way. What do you think?

Reply:
I fixed the issue with object editor success state. Atm, the snackbar component was isolated to the datastream features. If we want to utilize the Snackbar feature for the whole application, I would have to make some signficant modifications. Could we create a story for that task?

Reply:
Thanks for the fix. Do we need test coverage for this case?

Good idea regarding adding a story related to Snackbar refactoring. I've added one in Teams. I wouldn't give that high urgency yet, though -- but a reusable Snackbar could definitely be considered a future "quality of life" thing.

Reply:
Added unit tests.